### ISSUE TITLE
Add more working MQTT code samples/ references 

### ISSUE THREAD

**Issue by sacOO7** [ably/docs#1124]
- Extending https://github.com/ably/docs/issues/405
- Currently, official doc https://ably.com/documentation/mqtt, only contains JS specific sample code and doesn't show or point to other language specific code snippets.
- As a part of recent business use-cases, for IOT related snippets, we are using mqtt clients for each language, though there are no official references to those recommended libraries. 
- I would suggest either update MQTT doc on the website for each language specific working code snippet or update README for each of the SDK's. It should also include references to recommended MQTT third party client libraries.
- I would recommend going with second option since,
1. SDK's shouldn't just limit interfacing with respect to ably spec, rather it can include documentation about all related specs (in this case, MQTT spec), that expose interfaces and help connecting to ably server
2. Generally SDK README is what dev's refer to, to find standard API documentation

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/EDU-1745) by [Unito](https://www.unito.io)

**Comment by kavalerov** [comment]
> SDK's shouldn't just limit interfacing with respect to ably spec, rather it can include documentation about all related specs (in this case, MQTT spec), that expose interfaces and help connecting to ably server

How would Ably SDKs be involved with MQTT? As you mention elsewhere, alternative libraries would be used => I am not sure why documenting that behaviour would live in the Ably SDK readme.

Or am I misunderstanding something?

**Comment by sacOO7** [comment]
@kavalerov The first thing we need to agree on is, we need more comprehensible documentation for MQTT as I mentioned above, it can sit on various places. My pref. is to  put them inside README files of current SDK's (Like for python one, we have mentioned https://github.com/ably/ably-python#known-limitations). I don't see third party MQTT clients any different from current custom ably SDK implementation. In the end, these are libraries that help connect/ communicate with ably servers. If we are mentioning we support MQTT adapter, we should also mention which recommended MQTT clients to use, along with their code snippets, as we don't want people to try out unstable libs, which might fail to connect and devs can give up on using MQTT feature provided by ably. 
I feel this might be one of the most imp. thing we are currently missing on, if we already support MQTT, but still new ably users don't know how to use it properly. 
I agree MQTT supports minimal use-cases as compared to custom ably SDK's, but for most of the realtime apps, that's more than enough to get the job done.

**Comment by kavalerov** [comment]
I agree that we need more documentation, I just don't think it should belong to the readme's of Ably SDKs as it has nothing to do with those SDKs, and we want to keep readmes focused and concise. 

I agree though that we should have more extensive MQTT documentation in general, and then we can make sure that SDK Readmes link to the main documentation, to the relevant pages. This way MQTT documentation will be centralised, easier to maintain, and will not be mixed with any documentation for core Ably libraries.

**Comment by sacOO7** [comment]
Yes, links will surely help.
 But, I won't agree it has nothing to do with SDK's,
 User's will use MQTT, because current ably SDK protocol doesn't support it. Maybe it should be worth mentioning, SDK' use websocket protocol, for using MQTT protocol, visit this link

**Comment by kavalerov** [comment]
> Maybe it should be worth mentioning, SDK' use websocket protocol, for using MQTT protocol, visit this link

Exactly
